core/src/main/java/com/coderanch/blackjack/Card.java

```diff
@@ -10,33 +10,52 @@
 /**
  * A playing card used in a game of Blackjack.
  */
-final class Card {
-  
-  /**
-   * The rank of a {@link Card}.
-   */
-  enum Rank {}
+public final class Card {
 
-  /**
-   * The suit of a {@link Card}.
-   */
-  enum Suit {}
+  enum Rank {ONE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING, ACE};
+  enum Suit {SPADES, HEARTS, CLUBS, DIAMONDS};
   
-  /**
-   * Gets the rank of this card.
-   * 
-   * @return the rank of this card; never {@literal null}
-   */
-  Rank rank() {
-    return null;
+  // smcdonald4812 - value of the rank, secondary value for the ace, Rank, and Suit
+  private Rank rank;
+  private Suit suit;
+  //smcdonald4812 - could add a String ID here as well with UUID if wanted.
+  
+  //smcdonald4812 - default constructor, non-null assignment.
+  
+  public Card() {
```

Comment:
Having the one (ace?) of spades be the default seems rather arbitrary. I suggest removing the parameterless constructor so that the caller is always required to pass a rand and suit explicitly.

Reply:
After no-args (it isn't default, default is being added by Java when no constructor is defined explicitly) constructor remove, rank and suit fields can become final.